homeassistant/components/meteoswiss/coordinator.py

```diff
@@ -0,0 +1,54 @@
+"""DataUpdateCoordinator for Met.no integration."""
+
+import dataclasses
+import datetime
+import logging
+
+import meteoswiss_async
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, POSTAL_CODE, POSTAL_CODE_ADDITIONAL_NUMBER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass(kw_only=True, frozen=True)
+class MeteoSwissData:
+    """Data that gets updated by the coordinator."""
+
+    weather: meteoswiss_async.Weather
```

Comment:
I did it this way because there is another client endpoint that would need to be called in order to fetch data from stations, and that would require 2 different API calls and storing 2 data structures, and for that reason I created this dataclass to encapsulate it.